cora/scripts/makesky.py

```diff
@@ -190,28 +351,55 @@ def gaussianfg(ctx):
     fpol = galaxy.FullSkyPolarisedSynchrotron()
 
     # Set frequency parameters
-    fsyn.frequencies = ctx.obj.freq
+    fsyn.frequencies = fstate.frequencies
     nfreq = len(fsyn.frequencies)
 
-    nside = ctx.obj.nside
     lmax = 3 * nside
-    npol = 4 if ctx.obj.full_pol else 1
+    npol = 4 if pol == "full" else 1
 
-    cv_fg = np.zeros((lmax+1, npol, nfreq, npol, nfreq))
+    cv_fg = np.zeros((lmax + 1, npol, nfreq, npol, nfreq))
 
-    cv_fg[:, 0, :, 0, :] = skysim.clarray(fsyn.angular_powerspectrum, lmax, fsyn.nu_pixels)
+    cv_fg[:, 0, :, 0, :] = skysim.clarray(
+        fsyn.angular_powerspectrum, lmax, fsyn.nu_pixels
+    )
 
     if ctx.obj.full_pol:
-        cv_fg[:, 1, :, 1, :] = skysim.clarray(fpol.angular_powerspectrum, lmax, fsyn.nu_pixels)
-        cv_fg[:, 2, :, 2, :] = skysim.clarray(fpol.angular_powerspectrum, lmax, fsyn.nu_pixels)
-
-    cv_fg = cv_fg.reshape(lmax+1, npol*nfreq, npol*nfreq)
-
-    alms = skysim.mkfullsky(cv_fg, nside, alms=True).reshape(npol, nfreq, lmax+1, lmax+1)
+        cv_fg[:, 1, :, 1, :] = skysim.clarray(
+            fpol.angular_powerspectrum, lmax, fsyn.nu_pixels
+        )
+        cv_fg[:, 2, :, 2, :] = skysim.clarray(
+            fpol.angular_powerspectrum, lmax, fsyn.nu_pixels
+        )
+
+    cv_fg = cv_fg.reshape(lmax + 1, npol * nfreq, npol * nfreq)
+
+    alms = skysim.mkfullsky(cv_fg, nside, alms=True).reshape(
+        npol, nfreq, lmax + 1, lmax + 1
+    )
     alms = alms.transpose((1, 0, 2, 3))
 
     maps = hputil.sphtrans_inv_sky(alms, nside)
-    write_map(ctx.obj.filename, maps, fsyn.frequencies, ctx.obj.freq_width, ctx.obj.include_pol)
+    write_map(filename, maps, fsyn.frequencies, fstate.freq_width, pol != "none")
+
+
+@cli.command()
+@map_options
+@click.option("--ra", type=float, help="RA (in degrees) for source to add.", default=0)
+@click.option("--dec", type=float, help="DEC (in degrees) of source to add.", default=0)
+def singlesource(fstate, nside, pol, filename, ra, dec):
```

Comment:
Do we care that we put the point sources at locations as we please rather than 'real ps' postions? 
Do we want to have an option to change the amplitude?

Reply:
I wondered about the amplitude, but figured if you're only putting in one source then it doesn't matter.